python/lsst/cp/pipe/cpCombine.py

```diff
@@ -281,6 +281,13 @@ def run(self, inputExps, inputScales=None, inputDims=None):
         if numExps < self.config.maxVisitsToCalcErrorFromInputVariance:
             stats.setCalcErrorFromInputVariance(True)
 
+        detectorList = [exp.getDetector() for exp in inputExps]
+        if None in detectorList:
+            self.log.warn("Not all input detectors defined.")
+        numDetectors = len(set([det.getId() for det in detectorList if det is not None]))
+        if numDetectors > 1:
+            raise RuntimeError("Input data contains multiple detectors.")
```

Comment:
Presumably it's impossible for us to tell from a detector whether the detector itself is different even if the ID is the same? Do you care if the serial is different?

Reply:
I've written this comment arguing both ways (ids vs serials), and have finally just changed the code to check that the sets of both are the same length, and that that length is 1.